packages/block-editor/src/components/link-control/README.md

```diff
@@ -4,6 +4,26 @@ Renders a link control. A link control is a controlled input which maintains a v
 
 It is designed to provide a standardized UI for the creation of a link throughout the Editor, see History section at bottom for further background.
 
+## Best Practices
+
+### Ensuring unique instances
+
+It is possible that a given editor may render multiple instances of the `<LinkControl>` component. As a result, it is important to ensure each instance is unique across the editor to avoid state "leaking" between components.
+
+Why would this happen?
+
+React's reconciliation algorithm means that if you return the same element from a component, it keeps the nodes around as an optimization, even if the props change. This means that if you render two (or more) `<LinkControl>`s, it is possible that the `value` from one will appear in the other as you move between them.
```

Comment:
This could be slightly misleading. React only does that when the component node is at the same place in the tree. This happens in gutenberg because we use `SlotFill` to portal the component to different places, which in the meantime tricks the React reconciler into thinking that they are the same node.

Reply:
Questions:

- How would you recommend we improve this section of the README? 
- Shall we just remove this stuff about `key`? Or do you think it remains relevant (I think it is)?

Reply:
I don't have a preference TBH. I think it's okay to keep this, but maybe we can also make it less misleading. We could also add a link to the official React doc about reconciling for reference.